### PR TITLE
ARROW-4933: [R] Autodetect Parquet support using pkg-config

### PR DESCRIPTION
**Pull request by xhochy**
Kudos go to @kou for this.

### CODE REVIEW

**Review by wesm — APPROVED** [review]
+1

File: r/configure [hunk 1]
```diff
@@ -32,13 +32,26 @@ PKG_RPM_NAME="arrow"
...
 PKG_BREW_NAME="apache-arrow"
 PKG_TEST_HEADER="<arrow/api.h>"
-PKG_LIBS="-larrow -lparquet"
+PKG_LIBS=""
 
 # Use pkg-config if available
 pkg-config --version >/dev/null 2>&1
 if [ $? -eq 0 ]; then
   PKGCONFIG_CFLAGS=`pkg-config --cflags --silence-errors ${PKG_CONFIG_NAME}`
   PKGCONFIG_LIBS=`pkg-config --libs ${PKG_CONFIG_NAME}`
```
Comment (kou) [inline r/configure]: Needless codes are remained.
I've created a pull request: #3951

Comment (jeroen) [reply r/configure]: It's not needless code, it's a default on systems where `pkg-config` is unavailable. Also this PR has broken the build on MacOS. Did anyone actually test this?

Comment (kou) [reply r/configure]: Could you open a new JIRA issue for this?

**Pull request closed without merging**